### PR TITLE
fix: parser to throw multiple error messages only when `--continue-compilation` is enabled

### LINKED ISSUE #5332: Syntax errors now print all errors
It seems we now print all syntax errors, so we often get hundreds of error messages.

Instead, by default let's only print exactly one error message and exit. Only with `--continue-compilation` let's print all of them. We can also later control how many to print with a compiler option, but for now I think a good default is to print just one error message and exit.

Also, as a user, I don't think I want to run the code if there are errors. But I still might want to print all errors. We should figure out the CompilerOptions flags how to control this well.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Fixes #5332. While fixing this PR, I noticed that in https://github.com/lfortran/lfortran/pull/4774 we have "not" achieved error resiliency. That PR only collects error messages and dumps at the end, not skip statements. So the AST generated is incorrect. This means we cannot generate a valid ASR. We need to fix that.

### CODE REVIEW

File: tests/reference/run-multi_error1-2ef69ef.stderr [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  |
+5 | print *, x +
+  |             ^ 
+
+syntax error: Token 'y' (of type 'identifier') is unexpected here
+ --> tests/multi_error1.f90:6:12
+  |
+6 | print *, x y
+  |            ^ 
+clang: error: no such file or directory: 'tests/multi_error1.tmp.o'
```
Comment (Pranavchiku) [inline tests/reference/run-multi_error1-2ef69ef.stderr:24]: This is expected

File: tests/reference/run-multi_error1-2ef69ef.stdout [hunk 2]
```diff
@@ -0,0 +1 @@
+The command 'clang -o multi_error1.out tests/multi_error1.tmp.o  -L"$DIR/src/bin/../runtime" -Wl,-rpath,"$DIR/src/bin/../runtime" -llfortran_runtime -lm' failed.
```
Comment (Pranavchiku) [inline tests/reference/run-multi_error1-2ef69ef.stdout:1]: This is also expected because of the header comment in PR, AST is no valid and thus `ast.ok` results to false resulting in compiler not progress to compute ASR thus LLVM despite `--continue-compilation`.

Comment (Pranavchiku) [reply tests/reference/run-multi_error1-2ef69ef.stdout:1]: I mean these are bugs, which will be removed once we make parser truly error resilient.

File: tests/tests.toml [hunk 3]
```diff
@@ -4271,7 +4271,8 @@ run = true
 
 [[test]]
 filename = "multi_error1.f90"
-ast = true
+ast = true # TODO: remove this and only use continue compilation
+continue_compilation = true
```
Comment (Pranavchiku) [inline tests/tests.toml:4275]: once the above comments are addressed ( in subsequent PRs ) this will be converted to simply

```suggestion
continue_compilation = true
```

This will generate valid AST -> ASR -> LLVM -> binary -> execute it. RIght now AST is not valid.

File: run_tests.py [hunk 4]
```diff
@@ -136,14 +136,24 @@ def is_included(backend):
...
+                    extra_args)
+            else:
+                run_test(
+                    filename,
+                    "ast",
+                    "lfortran --show-ast --no-color {infile} -o {outfile}",
+                    filename,
+                    update_reference,
+                    verify_hash,
+                    extra_args)
```
Comment (Pranavchiku) [inline run_tests.py:156]: same comment as above, we can revert this change once we fix the above cases.

**Comment by Pranavchiku** [comment]
I opened https://github.com/lfortran/lfortran/issues/5344 for it.

File: tests/reference/run-multi_error1-2ef69ef.stderr (hunk 1 above)
Comment (certik) [reply tests/reference/run-multi_error1-2ef69ef.stderr:24]: Let's stop running the code, I think as a user I never want to run the code if there are errors.

File: tests/reference/run-multi_error1-2ef69ef.stdout (hunk 2 above)
Comment (certik) [reply tests/reference/run-multi_error1-2ef69ef.stdout:1]: Yes, we'll fix it later.

**Comment by certik** [comment]
@Pranavchiku let's get this merged, I need this to work on parser fixes.

**Comment by Pranavchiku** [comment]
> Let's stop running the code, I think as a user I never want to run the code if there are errors. So the default should be not to run it. We can add a separate flag to also run the code.

shall we merge this and then I can submit a patch to fix this? I am in middle of a code debug and will need a bit of back and froth.

**Comment by certik** [comment]
Yes, I think this is good to merge. Thanks!

**Comment by Pranavchiku** [comment]
I opened https://github.com/lfortran/lfortran/issues/5351 for it
